package_test.go

```diff
@@ -900,6 +900,84 @@ func (s *PackageSuite) TestTransactionErrors(c *C) {
 	c.Assert(err, IsNil)
 }
 
+func (s *PackageSuite) TestValidDML(c *C) {
+	var tests = []struct {
+		summary  string
+		query    string
+		types    []any
+		inputs   []any
+		outputs  [][]any
+		expected [][]any
+	}{{
+		summary:  "delete person",
+		query:    "DELETE FROM person WHERE id = $Person.id",
+		types:    []any{Person{}},
+		inputs:   []any{&Person{30, "Fred", 1000}},
+		outputs:  [][]any{},
+		expected: [][]any{},
+	}, {
+		summary:  "simple select person",
+		query:    "SELECT * AS &Person.* FROM person",
+		types:    []any{Person{}},
+		inputs:   []any{},
+		outputs:  [][]any{{&Person{}}, {&Person{}}, {&Person{}}, {&Person{}}},
+		expected: [][]any{{&Person{20, "Mark", 1500}}, {&Person{40, "Mary", 3500}}, {&Person{35, "James", 4500}}},
+	}, {
+		summary:  "insert person",
+		query:    "INSERT INTO person (*) VALUES ($Person.*)",
+		types:    []any{Person{}},
+		inputs:   []any{Person{50, "Jim", 6000}},
+		outputs:  [][]any{},
+		expected: [][]any{},
```

Comment:
SQLite as of 3.7.11 can also do bulk inserts of rows, we should check that this syntax is also available.

```sql
INSERT INTO person (name, age) VALUES ("fred", 12), ("matt", 32), ("alice", 42), ("tim", 14);
```

It might be worth figuring out what `$Person` would be in this case i.e. do we need to index them, or treat them like a stack?

```sql
INSERT INTO person (*) VALUES ($Person.*), ($Person.*), ($Person.*), ($Person.*);
```
or via the index syntax, which does allow for repetition (useful?). This could be backward compatible, with `$Person` being an alias to `$Person[0]`?

```sql
INSERT INTO person (*) VALUES ($Person[0].*), ($Person[1].*), ($Person[2].*), ($Person[0].*);
```

Reply:
If we do want to support bulk row updates, then the alternative syntax for bulk updates should have a test.

```
INSERT INTO person
          SELECT 'fred' AS "name", 12 AS "age"
UNION ALL SELECT 'alice', 14
UNION ALL SELECT 'bob', 15;
```

Reply:
I have actually been thinking this would be a pretty cool feature. The idea I had was that you would write your query as normal:
```sql
INSERT INTO person (*) VALUES ($Person.*)
```
And then when you do the query you could either pass a single `Person` OR a slice of Person: `db.Query(ctx, stmt, &personSlice)`. This would then expand the insert behind the scenes to be a bulk insert for all the Person objects in the slice. How does this sounds as an interface?

I also was thinking it would be a seperate PR after this got in.

Reply:
> How does this sounds as an interface?

This would be nice, I'm just slightly concerned about it not being obvious. Everything in sqlair feels obvious, this feels a bit magical, but don't hold that against this. 

Maybe `$Person.**` or something similar, i.e. expand everything. Then you can test if you get a `**` then it should be a slice.